jwst/resample/resample.py

```diff
@@ -1,31 +1,289 @@
 import logging
-import os
 import warnings
+import re
 
 import numpy as np
-import psutil
-from drizzle import cdrizzle, util
-from spherical_geometry.polygon import SphericalPolygon
+from astropy.io import fits
 
 from stdatamodels.jwst import datamodels
-from stdatamodels.jwst.library.basic_utils import bytes2human
-
-from jwst.datamodels import ModelContainer
-
-from . import gwcs_drizzle
-from jwst.resample import resample_utils
+from stcal.resample import ResampleModelIO, ResampleCoAdd, ResampleSingle
+from stcal.resample.utils import get_tmeasure
+from drizzle.resample import Drizzle
+from stdatamodels.jwst.datamodels.dqflags import pixel
+from stdatamodels.properties import ObjectNode
+
+from ..datamodels import ModelContainer
+from ..datamodels.library import ModelLibrary
 from ..model_blender import blendmeta
+from ..assign_wcs import util
+from .import resample_utils
 
 log = logging.getLogger(__name__)
 log.setLevel(logging.DEBUG)
 
-__all__ = ["OutputTooLargeError", "ResampleData"]
+__all__ = [
+    "OutputTooLargeError",
+    "ResampleJWSTModelIO",
+    "ResampleJWSTSingle",
+    "ResampleJWSTCoAdd",
+    "ResampleData",
+]
 
 
 class OutputTooLargeError(RuntimeError):
     """Raised when the output is too large for in-memory instantiation"""
 
 
+class ResampleJWSTModelIO(ResampleModelIO):
```

Comment:
Methods in `resample.py` are used in the `OutlierDetectionStep`, as well as the `ResampleSpecStep`.  Those need to be integrated into this PR

Reply:
> Methods in `resample.py` are used in the `OutlierDetectionStep`, as well as the `ResampleSpecStep`. Those need to be integrated into this PR

Yes, but this will be much easier to do after your PR that switches the pipeline to using `ModelLibrary` is merged.